Sources/PaymentServices/BasicPaymentService/BasicPaymentService.swift

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 Payoneer Germany GmbH
+// https://www.payoneer.com
+//
+// This file is open source and available under the MIT license.
+// See the LICENSE file for more information.
+
+import Foundation
+import Networking
+import Payment
+import UIKit
+
+@objc public class BasicPaymentService: NSObject, PaymentService {
+    // MARK: - Static methods
+    public static func isSupported(networkCode: String, paymentMethod: String?) -> Bool {
+        if let paymentMethod = paymentMethod {
+            if isSupported(method: paymentMethod) { return true }
+        }
+
+        if isSupported(code: networkCode) { return true }
+
+        return false
+    }
+
+    private static func isSupported(method: String) -> Bool {
+        let supportedMethods: [PaymentMethod] = [.DEBIT_CARD, .CREDIT_CARD]
+        guard let paymentMethod = PaymentMethod(rawValue: method) else {
+            return false
+        }
+
+        return supportedMethods.contains(paymentMethod)
+    }
+
+    private static func isSupported(code: String) -> Bool {
+        let supportedCodes = ["SEPADD", "PAYPAL", "WECHATPC-R"]
+        return supportedCodes.contains(code)
+    }
```

Comment:
You could turn the three functions into one:

```suggestion
    public static func isSupported(networkCode: String, paymentMethod: String?) -> Bool {
        if let methodString = paymentMethod, let method = PaymentMethod(rawValue: methodString) {
            let supportedMethods: [PaymentMethod] = [.DEBIT_CARD, .CREDIT_CARD]
            return supportedMethods.contains(method)
        } else {
            let supportedCodes = ["SEPADD", "PAYPAL", "WECHATPC-R"]
            return supportedCodes.contains(network)
        }
    }
```

Reply:
Changed in `c8096c8a32dccc1346aedc414b9c2852e3c18188`